pkg/cloud/azure/actuators/machine/actuator.go

```diff
@@ -70,18 +98,20 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 		CoreClient: a.coreClient,
 	})
 	if err != nil {
-		return errors.Errorf("failed to create scope: %+v", err)
+		return a.handleMachineError(machine, apierrors.CreateMachine("failed to create scope: %v", err), createEventAction)
 	}
 	defer scope.Close()
 
-	err = NewReconciler(scope).Create(context.Background())
+	err = a.reconcilerBuilder(scope).Create(context.Background())
 	if err != nil {
-		klog.Errorf("failed to reconcile machine %s: %v", machine.Name, err)
+		a.handleMachineError(machine, apierrors.CreateMachine("failed to reconcile machine %s: %v", machine.Name, err), createEventAction)
 		return &controllerError.RequeueAfterError{
 			RequeueAfter: time.Minute,
 		}
 	}
 
+	a.eventRecorder.Eventf(machine, corev1.EventTypeNormal, "Created", "Created machine %v", machine.Name)
```

Comment:
nitL: can we use either %s, %v or %q consitently for priting names? my preference is %q